src/v/storage/segment_set.cc

```diff
@@ -468,9 +467,9 @@ ss::future<segment_set> recover_segments(
       })
       .then([&as,
              is_compaction_enabled,
-             last_clean_segment = std::move(last_clean_segment),
-             is_internal_topic](segment_set::underlying_t segs) {
-          auto segments = segment_set(std::move(segs), is_internal_topic);
+             last_clean_segment = std::move(last_clean_segment)](
+              segment_set::underlying_t segs) {
+          auto segments = segment_set(std::move(segs));
```

Comment:
hm, it looked like this plumbing was useful in the previous commit. did you mean to squash this back into the previous commit?